cluster/mutex.go

```diff
@@ -58,47 +52,109 @@ type Mutex struct {
 	lock        sync.Mutex
 	stopRefresh chan bool
 	refreshDone chan bool
+
+	// lockExpiry tracks the expiration time of the lock when last locked. It is not guarded
+	// by a local mutex, since it is only read or written when the cluster lock is held.
+	lockExpiry time.Time
 }
 
-// NewMutex creates a mutex with the given name.
+// NewMutex creates a mutex with the given key name.
+//
+// Panics if key is empty.
 func NewMutex(pluginAPI MutexPluginAPI, key string) *Mutex {
-	key = mutexPrefix + key
-
 	return &Mutex{
 		pluginAPI: pluginAPI,
-		key:       key,
+		key:       makeLockKey(key),
 	}
 }
 
+// makeLockKey returns the prefixed key used to namespace mutex keys.
+func makeLockKey(key string) string {
+	if len(key) == 0 {
```

Comment:
0/5 return a "" which is bound to fail downstream. `panic` in production code seems like something to avoid?

Reply:
My original rationale for panicking here was to make the `NewMutex` constructor not have to return an `error`, effectively simplifying the interface and ultimately making it just a little more accessible (and more like `sync.Mutex`). But your comment reminded me to vet this thinking, and I came across this clear directive from https://blog.golang.org/defer-panic-and-recover:

> The convention in the Go libraries is that even when a package uses panic internally, its external API still presents explicit error return values.

I'll change this back -- thank you!